qui/tray/devices.py

```diff
@@ -235,6 +262,11 @@ def device_list_update(self, vm, _event, **_kwargs):
                     notification_id=(dev.backend_domain +
                                      dev.ident))
 
+        for devclass in DEV_TYPES:
+            for device in vm.devices[devclass].attached():
```

Comment:
So a minimal change of these four lines seems to be required to get the UI to update even with your diff in https://github.com/QubesOS/qubes-app-linux-usb-proxy/pull/15/files @marmarek

Above on line 248 when we iterate through `vm.devices[devclass].available()`, the device that has been attached (using a udev rule) does not appear, however that device _does_ appear in `vm.devices[devclass].attached()` - is that expected behavior? 

I'm asking because I'm wondering if that is an issue with the UI logic here or qubesadmin or otherwise

Reply:
`.available()` is about devices exposed by a given backend VM - normally `sys-usb`. `.attached()` is about devices connected to a given frontend VM. So, it is normal that device does not appear in both `.attached()` and `.available()` of the same VM - should be visible on different VMs: `.available()`  of `sys-usb` and `.attached()` of the VM where you connect the device.

> he device that has been attached (using a udev rule) does not appear

You mean isn't marked as connected, but is present in the menu, right?

Reply:
ahh scratch this, i forgot to restart qubesd ;)